### ISSUE TITLE
File containing a single unescaped " out-of-sample is read incorrectly

### ISSUE THREAD

**Issue by st-pasha** [h2oai/datatable#1036]
When `fill=False`, this example raises an error:
```
>>> src = "A,B,C\n" + "q,f,r\n" * 100 + "foo,\"bar,bza\n" + "a,bb,ccc\n" * 200
>>> dt.fread(src, fill=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/pasha/github/datatable/datatable/fread.py", line 69, in fread
    return freader.read()
  File "/Users/pasha/github/datatable/datatable/fread.py", line 715, in read
    _dt = core.gread(self)
RuntimeError: Too few fields on line 102: expected 3 but found only 2 (with sep=','). Set fill=True to ignore this error.  <<foo,"bar,bza>>
```
even though the file can be read with QR=3 (and in fact if we move problematic line to the beginning of the file, it will be parsed correctly).

When `fill=True` the file does not throw an error, but returns a 101x3 Frame where the last row contains value `"bar,bza\na,bb,ccc\n...` with the content of the rest of the file.

In addition, verbose mode shows
```
  Estimated number of rows: 2419 / 6.00 = 404
  Initial alloc = 444 rows (404 + 9%) using bytes/max(mean-2*sd,min) clamped between [1.1*estn, 2.0*estn]
  All rows were sampled since file is small so we know nrows=99 exactly
```
Obviously not all rows were sampled, and therefore `nrows=99` is a gross underestimation (there should be 301 rows).

**Comment by pradkrish** [comment]
When I run this example on main branch with `fill=True,` it does return a 301*3 frame. 

By the way, how did you run in it verbose mode? I did `python -v filename.py` and did not see the info you mentioned above.

**Comment by st-pasha** [comment]
"verbose mode" refers to supplying parameter `verbose=True` to fread:
```
dt.fread(src, fill=False, verbose=True)
```

**Comment by st-pasha** [comment]
Indeed, this example seem to be working correctly now. So closing the issue.

**Comment by st-pasha** [comment]
Ok, I just tried creating a test example for this, and it turns out it's not quite working.
See `test-fread-issues.py::test_issue1036`.

**Comment by pradkrish** [comment]
@st-pasha I don't see `test-fread-issues.py::test_issue1036` on the main branch, are you sure you have added that example?

**Comment by st-pasha** [comment]
You're right, I forgot to push that branch earlier. See #2730

**Comment by st-pasha** [comment]
Closed in #2709
